RFC: detect closed port

jsiverskog opened the pull request:
if a port disappeared and reappeared quickly, while someone was
awaiting receive_some, the receive_some loop ended as an infinite loop
consuming 100% cpu.

note: this should probably be done one level up, in `receive_some`.

i'm just throwing it out there to get some feedback. i'm not sure this is a safe way of detecting the issue.

joernheissler commented:
Hi,
thanks for the report.
It's easy to reproduce by simply removing the serial device; no need to reconnect it.
Looks like I never considered the possibility of an EOF on a serial stream. Your idea with the `stat` would work on Linux (at least on recent versions). Sadly I can't test this on other posix systems.
Pyserial raises an exception on empty read, so perhaps that's what should be checked for here, too.

jsiverskog commented:
@joernheissler right. i have access to mac(s) at least, so i can test on that platform.

doing it like pyserial does it probably makes sense (if empty read is a good and safe way of detecting this).

jsiverskog commented:
hey, any ideas on this?

thanks.

jsiverskog commented:
hey,
still having issues because of this. any updates?

joernheissler commented:
I can't remember why I created the loop in the first place. Perhaps the correct fix would be remove the loop and let the caller handle the situation.

jsiverskog committed:
remove infinite loop

if a port disappeared and reappeared quickly, while someone was
awaiting receive_some, the receive_some loop ended as an infinite loop
consuming 100% cpu. i can't figure out what this loop is good for.

jsiverskog commented:
@joernheissler : i (briefly) tested this now with the loop removed and it seems to work. i have updated the commit.

sorry for the late response.

jsiverskog commented:
@joernheissler : it would be great to merge this and get a new version released. thanks.
